tasks/yarn.yml

```diff
@@ -47,23 +48,49 @@
     mode: g=rXw,o=rX
     recurse: true
 
+- name: set permissions to client
+  file:
+    dest: "{{ wifi_login_pages_path }}/client"
+    group: "{{ www_group }}"
+    # sets executable only to directories
+    mode: u=rwX,g=rX,o=rX
+    recurse: true
+
+- name: set permissions to server
+  file:
+    dest: "{{ wifi_login_pages_path }}/server"
+    group: "{{ www_group }}"
+    # sets executable only to directories
+    mode: u=rwX,g=rX,o=rX
+    recurse: true
+
+- name: copy organizations configs and assets
+  become: true
+  copy:
+    src: owlp_organizations/
+    dest: "{{ wifi_login_pages_path }}/organizations"
+    owner: "{{ www_user }}"
+    group: "{{ www_group }}"
+    mode: 0644
+
 - name: yarn build
   become: true
-  become_user: www-data
   command: "yarn build"
   args:
     chdir: "{{ wifi_login_pages_path }}"
 
-- name: set permissions to server config
+- name: change permissions of client
   file:
-    dest: "{{ wifi_login_pages_path }}/server/config.json"
+    dest: "{{ wifi_login_pages_path }}/client"
     owner: "{{ www_user }}"
     group: "{{ www_group }}"
     mode: 0660
+    recurse: true
 
-- name: set permissions to client config
+- name: change permissions of server
   file:
-    dest: "{{ wifi_login_pages_path }}/client/config.json"
+    dest: "{{ wifi_login_pages_path }}/server"
     owner: "{{ www_user }}"
     group: "{{ www_group }}"
     mode: 0660
+    recurse: true
```

Comment:
why did you change these two tasks?

I only noticed now that the system where I tested this now is showing permission issues.

Reply:
This is the script I am using to rebuild the app:

```
#!/bin/bash
set -e
yarn
# everything owned by www-data
chown -R www-data:www-data .
# remove any public permission
chmod -R o-rwx .
# build in production mode as www-data
sudo -u www-data yarn build --mode production
# restart
supervisorctl restart wifi_login_pages
```

I think we should use something similar here:

- there's no need to keep any public permission
- ensure www-data owns all files
- do not touch any other permission